build/common.sh

```diff
@@ -380,7 +329,6 @@ function kube::build::destroy_container() {
 function kube::build::clean() {
   if kube::build::has_docker ; then
     kube::build::docker_delete_old_containers "${KUBE_BUILD_CONTAINER_NAME_BASE}"
-    kube::build::docker_delete_old_images "${KUBE_BUILD_IMAGE_REPO}" "${KUBE_BUILD_IMAGE_TAG_BASE}"
```

Comment:
If you check the behavior above, this meant we deleted old KUBE_BUILD images that did not have the current tag, but we have stopped building kube-build images so this is wasted effort. It did not clean up any other images.

Developers will need to prune unused docker images anyhow. Dropping this speeds up `make clean`

`make clean` on any version that built a kube-build image will still clean it up.